tensorbay/opendataset/nuImages/loader.py

```diff
@@ -0,0 +1,235 @@
+    segment: FusionSegment,
+    sensor_name: str,
+    calibrated_sensor_info: Dict[str, Any],
+) -> None:
+    sensor = Camera(sensor_name)
+    sensor.set_extrinsics(
+        translation=calibrated_sensor_info["translation"],
+        rotation=calibrated_sensor_info["rotation"],
+    )
+    intrinsics = calibrated_sensor_info["camera_intrinsic"]
+
+    if intrinsics:
+        sensor.set_camera_matrix(matrix=intrinsics)
+        distortion = calibrated_sensor_info["camera_distortion"]
+        if distortion:
+            distortion_keys = ("k1", "k2", "p1", "p2", "k3", "k4")
+            kwargs = dict(zip(distortion_keys, distortion))
+            sensor.set_distortion_coefficients(**kwargs)
+    segment.sensors.add(sensor)
+
+
+def _load_labels(
+    data: Data,
+    sample_data_token: str,
+    annotation_info: Dict[str, Dict[str, Any]],
+) -> None:
+    object_annotations = annotation_info["object_annotations"]
+    surface_annotations = annotation_info["surface_annotations"]
+    categories = annotation_info["category"]
+    attributes = annotation_info["attribute"]
+    box2d: List[LabeledBox2D] = []
+    rle: List[LabeledRLE] = []
+    if sample_data_token in object_annotations:
+        _set_object_annotations(
+            object_annotations[sample_data_token], categories, attributes, box2d, rle
+        )
+    if sample_data_token in surface_annotations:
+        _set_surface_annotations(surface_annotations[sample_data_token], categories, rle)
+    data.label.box2d = box2d
+    data.label.rle = rle
+
+
+def _set_object_annotations(
+    object_annotations: List[Dict[str, Any]],
+    categories: Dict[str, Any],
+    attributes: Dict[str, Any],
+    box2d: List[LabeledBox2D],
+    rle: List[LabeledRLE],
+) -> None:
+    for object_annotation in object_annotations:
+        category = categories[object_annotation["category_token"]]["name"]
+        attributes_label = {}
+        for attribute_token in object_annotation["attribute_tokens"]:
+            attribute_names = attributes[attribute_token]["name"].split(".", 1)
+            attributes_label[attribute_names[0]] = attribute_names[1]
+        box2d.append(
+            LabeledBox2D(*object_annotation["bbox"], category=category, attributes=attributes_label)
+        )
+        mask = object_annotation["mask"]
+        if mask:
+            rle.append(
+                LabeledRLE(
+                    uncompress_rle(
+                        bytes.decode(base64.b64decode(mask["counts"])),
+                        mask["size"][0],
+                        mask["size"][1],
+                    ),
+                    category=category,
+                    attributes=attributes_label,
+                )
+            )
+
+
+def _set_surface_annotations(
+    surface_annotations: List[Dict[str, Any]], categories: Dict[str, Any], rle: List[LabeledRLE]
+) -> None:
+    for surface_annotation in surface_annotations:
+        mask = surface_annotation["mask"]
+        if mask:
```

Comment:
The code block after `if mask` is similar with the code block in `_set_object_annotations`, should we wrap it?